### PR TITLE
Issue 158: Added TemplateFormulaCell

### LINKED ISSUE #158: Add TemplateFormulaCell : Possibility to add a template for a custom formula
As @jasonclake requested on a [Gitter channel](https://gitter.im/Siemienik/community?at=60235fb7ae4b9b27c1aeff31)

image

>@Siemienik Is there a way to template a custom formula
I have a loop starting on row 2 running to row 6 (see image)

Transform something like

```
#! =SUMPRODUCT( B${test.__startOutput.r}:B${test.__endOutput.r}, C${test.__endOutput.r}:C${test.__endOutput.r}) / SUM(B${test.__startOutput.r}:B${test.__endOutput.r})
```

Into:

```
=SUMPRODUCT(B2:B6,C2:C6)/SUM(B2:B6)
```

### PR DESCRIPTION
**Pull request by Siemienik**
resolves #158 

image

### CODE REVIEW

File: packages/xlsx-renderer/README.md [hunk 1]
```diff
@@ -64,20 +64,21 @@ It is possible to use the command line interface [read more about xlsx-renderer-
...
+| Content | [HyperlinkCell](./src/cell/HyperlinkCell.ts) | 6 | `#! HYPERLINK pathToLabel pathToTarget` | Create a hyperlink. | *Paths resolve exactly same as VariableCell* |
+| Content | [FormulaCell](./src/cell/FormulaCell.ts) | 5 | Cell.type eq. formulae | It handles correctly formulas inside and outside of loops - when rows were shifted compared to the template. | *It is used automatically when formulae from the template being rendered* <br/> [Example](./tests/integration/data/Renderer010-Formula)|
+| Content | [TemplateFormulaCell](./src/cell/TemplateFormulaCell.ts) | 4 | Starts with `#= ` | This one allows you to put a template string into a cell as a formula. To write in a variable use `${patToVariable}`. | **Example:**<br/> `#= ${summaryFormula}(A2:A${tem.__endOutput.r})` gives something like `=MAX(A2:A2910)` <br/> [Example](./tests/integration/data/Renderer017-TemplateFormula)|
```
Comment (Siemienik) [inline packages/xlsx-renderer/README.md:71]: This line was added, others lines have only changed `Matching Order`

Comment (jasonclake) [inline packages/xlsx-renderer/README.md:71]: ## very minor typos
* Missing h ${patToVariable}
* Missing i (A2:A${tem.__endOutput.r})

## Later in the examples list you are missing the new id 17 example

**Review by jasonclake — APPROVED** [review]
Looks good!  Very minor typos noted (just proving that I read through your work :) opens up a lot of flexibility

File: packages/xlsx-renderer/README.md (hunk 1 above)
Comment (Siemienik) [reply packages/xlsx-renderer/README.md]: 🤦 Thank you for catching that

**Pull request merged**